sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala

```diff
@@ -175,7 +187,13 @@ private[sql] class JSONOptions(
     parameters.get(WRITE_NON_ASCII_CHARACTER_AS_CODEPOINT).map(_.toBoolean).getOrElse(false)
 
   /** Build a Jackson [[JsonFactory]] using JSON options. */
-  def buildJsonFactory(): JsonFactory = {
+  def buildJsonFactory(maxNestingDepth: Int, maxNumLen: Int, maxStringLen: Int): JsonFactory = {
```

Comment:
I don't think it's necessary to add them to the parameter list of `buildJsonFactory`

Reply:
I want to know if `streamReadConstraints` will throw exceptions that have not been handled?

Reply:
IllegalArgumentExceptions can be thrown when configuring StreamReadConstraints (negative values for lengths, etc) - https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-core/latest/com/fasterxml/jackson/core/StreamReadConstraints.Builder.html